tests/integration/olp-cpp-sdk-dataservice-read/VersionedLayerPrefetchPartitionsTest.cpp

```diff
@@ -0,0 +1,344 @@
+/*
+ * Copyright (C) 2020 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#include <gmock/gmock.h>
+
+#include <algorithm>
+#include <chrono>
+#include <memory>
+#include <numeric>
+#include <string>
+
+#include <matchers/NetworkUrlMatchers.h>
+#include <mocks/NetworkMock.h>
+#include <olp/authentication/Settings.h>
+#include <olp/core/cache/CacheSettings.h>
+#include <olp/core/client/OlpClientSettings.h>
+#include <olp/core/client/OlpClientSettingsFactory.h>
+#include <olp/dataservice/read/VersionedLayerClient.h>
+#include <olp/dataservice/read/model/Partitions.h>
+#include "HttpResponses.h"
+#include "PlatformUrlsGenerator.h"
+#include "ReadDefaultResponses.h"
+#include "ResponseGenerator.h"
+#include "VersionedLayerTestBase.h"
+
+using mockserver::ReadDefaultResponses;
+using olp::http::HttpStatusCode;
+using testing::_;
+namespace {
+
+namespace read = olp::dataservice::read;
+namespace client = olp::client;
+namespace http = olp::http;
+
+constexpr auto kTimeout = std::chrono::seconds(3);
+
+class VersionedLayerPrefetchPartitionsTest : public VersionedLayerTestBase {
+ protected:
+  std::vector<std::string> GeneratePartitionIds(size_t partitions_count) {
+    std::vector<std::string> partitions;
+    partitions.reserve(partitions_count);
+    for (auto i = 0u; i < partitions_count; i++) {
+      partitions.emplace_back(std::to_string(i));
+    }
+    return partitions;
+  }
+
+ protected:
+  const std::string layer_ = "testlayer";
+};
```

Comment:
This is not something that is really worth using an extra class for.
I would just use `using VersionedLayerClientPrefetchPartitionsTest = VersionedLayerTestBase;` and make the function anonymous.